### PR TITLE
JENKINS-45589 check for null when using OrganizationFactory

### PR DESCRIPTION
**Pull request by i386**
# Description

It's possible not to get back a valid `BlueOrganization` when using methods on `OrganizationFactory`. There still feels like there is a bug somewhere here but without paths for the items that we are resolving there is no way to tell.

This should add some better exception messages and setup expectations that `OrganizationFactory` may not work in some circumstances.

See [JENKINS-45589](https://issues.jenkins-ci.org/browse/JENKINS-45589).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by i386** [comment]
Expecting we may get some findbugs violations

**Comment by i386** [comment]
@vivek you may want to take a look at this again

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineImpl.java [hunk 1]
```diff
@@ -18,17 +20,18 @@
...
     }
 
     @Extension(ordinal = 1)
     public static class PipelineFactoryImpl extends BluePipelineFactory {
 
         @Override
         public BluePipeline getPipeline(Item item, Reachable parent) {
-            if (item instanceof WorkflowJob) {
-                return new PipelineImpl((Job) item);
+            BlueOrganization org = OrganizationFactory.getInstance().getContainingOrg(item);
```
Comment (i386) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineImpl.java]: We should never be doing this lookup inside of the CTOR for the Pipeline Impls...

Comment (vivek) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineImpl.java]: Yeah, better to pass them thru constructor param.

**Comment by michaelneale** [comment]
:bee:

**Pull request merged**